src/core/lib/transport/metadata_batch.h

```diff
@@ -45,6 +45,21 @@
 
 namespace grpc_core {
 
+template <typename Trait, typename Ignored = void>
+struct IsEncodableTraitImpl {
```

Comment:
Please add comments for the new code here.

Reply:
will do once i've satisfied the older compilers that this is c++

Reply:
done